packages/calling-component-bindings/src/handlers/createTeamsCallHandlers.ts

```diff
@@ -0,0 +1,121 @@
+ * See {@link useHandlers} and {@link usePropsFor}.
+ *
+ * @beta
+ */
+export type TeamsCallingHandlers = Omit<CallingHandlersCommon, 'onStartCall'> & {
+  onStartCall: (
+    participants: CommunicationIdentifier[],
+    options?: StartCallOptions
+  ) => void | /* @conditional-compile-remove(teams-identity-support) */ TeamsCall;
+};
+
+/**
+ * Create the default implementation of {@link CallingHandlers} for teams call.
+ *
+ * Useful when implementing a custom component that utilizes the providers
+ * exported from this library.
+ *
+ * @beta
+ */
+export const createDefaultTeamsCallingHandlers = memoizeOne(
+  (
+    callClient: StatefulCallClient,
+    callAgent: undefined | /* @conditional-compile-remove(teams-identity-support) */ TeamsCallAgent,
+    deviceManager: StatefulDeviceManager | undefined,
+    call: undefined | /* @conditional-compile-remove(teams-identity-support) */ TeamsCall
+  ): never | TeamsCallingHandlers => {
+    const calingsHandlers = createDefaultCallingHandlersCommon(callClient, deviceManager, call);
+    return {
+      ...calingsHandlers,
+      onStartCall: (participants, options) => {
+        /* @conditional-compile-remove(teams-identity-support) */
+        const threadId = options?.threadId;
+        if (!isTeamsCallParticipants(participants)) {
+          throw new Error('CommunicationIdentifier in Teams call is not supported!');
+        }
+        /* @conditional-compile-remove(teams-identity-support) */
+        if (callAgent) {
+          return callAgent.startCall(participants, threadId ? { threadId } : undefined);
+        }
+        return undefined;
+      },
+      /* @conditional-compile-remove(teams-identity-support) */
+      /* @conditional-compile-remove(PSTN-calls) */
+      onAddParticipant: async (
+        userId: string | CommunicationIdentifier,
+        options?: AddPhoneNumberOptions
+      ): Promise<void> => {
+        const participant = _toCommunicationIdentifier(userId);
+        /* @conditional-compile-remove(teams-identity-support) */
+        const threadId = options?.threadId;
+        /* @conditional-compile-remove(teams-identity-support) */
+        if (isPhoneNumberIdentifier(participant)) {
+          call?.addParticipant(participant, threadId ? { threadId } : undefined);
+        } else {
+          if (isCommunicationUserIdentifier(participant)) {
+            throw new Error('CommunicationIdentifier in Teams call is not supported!');
+          }
+          /* @conditional-compile-remove(teams-identity-support) */
+          call?.addParticipant(participant, threadId ? { threadId } : undefined);
+        }
+      },
+      onRemoveParticipant: async (
+        userId: string | /* @conditional-compile-remove(PSTN-calls) */ CommunicationIdentifier
+      ): Promise<void> => {
+        const participant = _toCommunicationIdentifier(userId);
+        if (isCommunicationUserIdentifier(participant)) {
```

Comment:
Same question as above. Why does this check exist in the UI layer as opposed to the headless layer?

Reply:
Because we want to align handler signature to simplify component logic for now - only one or two identifier types are not worthy to create an entirely new set of handlers